### ISSUE TITLE
B_박성빈_lab_week3

### ISSUE THREAD

**Issue by 03psb** [vlab-kaist/NeuralNet101#23]
### LAB 3 - 박성빈
**Problem 1**

    import torch
    import torch.nn.functional as F
    import pandas as pd

    df = pd.read_csv("Breast_cancer_train.csv")
    x_train = torch.tensor(df.loc[:, ['mean_radius', 'mean_texture', 'mean_perimeter', 'mean_area', 'mean_smoothness']].values.tolist(), 
    dtype=torch.float32)
    y_train = torch.tensor(df['diagnosis'].values.tolist(), dtype=torch.float32).reshape(-1, 1)

    W = torch.zeros((5, 1), requires_grad=True, dtype=torch.float32)
    b = torch.zeros(1, requires_grad=True, dtype=torch.float32)

    optimizer = torch.optim.SGD([W, b], lr=0.001)

    nb_epochs = 10000
    for epoch in range(nb_epochs):
        hypothesis = torch.sigmoid(x_train.matmul(W) + b)
        cost = F.binary_cross_entropy(hypothesis, y_train)

        optimizer.zero_grad()
        cost.backward()
        optimizer.step()

        if epoch%100 == 0:
            print(cost.item())

    x_predict = torch.tensor([[17.99,10.38,122.8,1001,0.1184],[13.54,14.36,87.46,566.3,0.09779]], dtype=torch.float32)
    y_predict = torch.sigmoid(x_predict.matmul(W) + b)
    print(y_predict)
    prediction = (y_predict >= torch.Tensor([0.5])).int()
    print(prediction)

**output**

    tensor([[1.],
            [1.]], grad_fn=<SigmoidBackward0>)
    tensor([[1],
            [1]], dtype=torch.int32)

**Problem 2**

    import pandas as pd
    import torch
    import torch.nn as nn
    import torch.nn.functional as F

    class BinaryClassifier(nn.Module):
        def __init__(self):
            super().__init__()
            self.linear = nn.Linear(14, 1)
            self.sigmoid = nn.Sigmoid()

        def forward(self, x):
            return self.sigmoid(self.linear(x))

    model = BinaryClassifier()
    optimizer = torch.optim.SGD(model.parameters(), lr=0.01)

    df = pd.read_csv("titanic_train.csv")
    x_train = torch.tensor(df.loc[:, ['Sex', 'Age', 'Fare', 'Pclass_1', 'Pclass_2', 'Pclass_3', 'Family_size', 'Title_1', 'Title_2', 'Title_3', 'Title_4', 
    'Emb_1', 'Emb_2', 'Emb_3']].values.tolist(), dtype=torch.float32)
    y_train = torch.tensor(df['Survived'].values.tolist(), dtype=torch.float32).reshape(-1, 1)

    nb_epochs = 100000
    for epoch in range(nb_epochs):
        hypothesis = model(x_train)
        cost = F.binary_cross_entropy(hypothesis, y_train)

        optimizer.zero_grad()
        cost.backward()
        optimizer.step()

        if epoch%100==0:
            print(cost.item())
